0-common/igluctl/src/main/scala/com.snowplowanalytics.iglu/ctl/commands/Pull.scala

```diff
@@ -92,12 +89,20 @@ object Pull {
       Either.catchNonFatal(request.asString).leftMap(_.getMessage)
     }).flatMap { response =>
       if (response.code == 200) EitherT.pure[IO, String](response)
-      else EitherT.leftT[IO, HttpResponse[String]]("Status code is not 200.")
+      else EitherT.leftT[IO, HttpResponse[String]](s"Unexpected status code ${response.code}. Response body: ${response.body}.")
     }
 
-  def writeSchemas(schemas: HttpResponse[String], output: Path): EitherT[IO, Common.Error, List[String]] = {
-    val parsed = jacksonParse(schemas.body).extract[List[JValue]].map(igluParse[JValue](_).getOrElse(throw new RuntimeException("Invalid self-describing JSON schema"))).map {
-      schema => textFile(Paths.get(s"$output/${schema.self.schemaKey.toPath}"), schema.asString)
+  /**
+    * Parse the response from Iglu Server and write it under the specified output path.
+    * Perorms IO
+    *
+    * @param response An [[HttpResponse]] returned by Iglu Server
+    * @param outputDir The path to write to
+    * @return A list of IO commands or Errors (if a schema fails validation), wrapped in an [[EitherT]]
+    */
+  def writeResponseAsSchemas(response: HttpResponse[String], outputDir: Path): EitherT[IO, Common.Error, List[String]] = {
```

Comment:
Hey @dilyand, do you want to improve this function? There are couple of things I see:

1. Exception. This can be just `Left`
2. `IO`. This function does two things: parsing `HttpResponse` into schemas and writing data to disk. Since it writes something to disk we can test it only if we write something to disk as well, which means it is not that pure. But what if we'll have two functions: one to parse `HttpResponse[String]` into `List[SelfDescribingSchema]` (result type will be slightly more complex) and another to to write it to disk? Signature of first function will have no `IO`, thus we always know that function is pure and can be tested.

If you want to do this - I'd recommend to address second issue first.

Reply:
Done.